### PR TITLE
fixes Issue#427

### PR DESCRIPTION
**Pull request by ricardogsilva**
# Overview

This Pull request makes tests more robust by comparing XML documents by using canonicalization instead of doing a simple string comparison. JSON tests are also compared by using converting python dicts to sorted lists of tuples.

# Related Issue / Discussion

More information at #427 

# Additional Information

This PR is not complete yet. I still have to merge the more recent changes in. There are also 5 tests that are failing. These are all related to the SRU suite and I believe they fail for an unrelated issue. This still needs to be clarified though.

# Contributions and Licensing

(as per https://github.com/geopython/pycsw/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pycsw. I confirm that my contributions to pycsw will be compatible with the pycsw license guidelines at the time of contribution.
- [x] I have already previously agreed to the pycsw Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by ricardogsilva** [comment]
@tomkralidis There are currently five tests that fail, but I believe them to be unrelated to this PR. They come from the fact that the prefix used in SRU requests seems to have changed recently. The test suite is expecting an `sru` prefix for the `http://www.loc.gov/zing/srw/` namespace, but the server is replying with a `zs` prefix to this namespace. Any thoughts on this?

I'll merge the recent changes to the master branch (the elapsed time stuff) into this PR and clean up a bit and it should become ready for reviewing.

**Comment by tomkralidis** [comment]
Sounds good @ricardogsilva. Let's look deeper once travis is able to run.

**Comment by ricardogsilva** [comment]
@tomkralidis it seems that all tests are passing in Travis now, which is a bit strange as I get 5 failed tests on my dev macchine when running under python2.7 (they all pass with python3 though). What could be the cause for the differences? Something to do with the testing database?

Another thing, I've left the older code, that performs the comparison using `cmplib.cmp`, in the file, still available by using a Pedantic flag, as @tomkralidis suggested. Personally I'm not a big fan of this. I'd rather just replace the old implementation with this one, because I'm thinking that the pedantic flag will end up not being used much (or at all). I'll let you decide.

**Comment by tomkralidis** [comment]
@ricardogsilva can you squash your commits?

File: tests/run_tests.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # =================================================================
 #
 # Authors: Tom Kralidis <tomkralidis@gmail.com>
+#          Ricardo Garcia Silva <ricardo.garcia.silva@gmail.com>
 #
 # Copyright (c) 2016 Tom Kralidis
```
Comment (tomkralidis) [inline tests/run_tests.py]: Suggest adding your name here as well, i.e.:

```
Copyright (c) 2016 Ricardo Garcia Silva
```

**Comment by ricardogsilva** [comment]
@tomkralidis I can't seem to squash cleanly. Maybe I shouldn't have made the merge with master before squashing... I've tried different ways, but I keep getting conflicts. I've got to practice rebasing more ;)
Can this go through as is?

**Comment by tomkralidis** [comment]
@ricardogsilva actually GitHub gave the ability to squash and merge (nice!).  Thanks for this valuable contribution!

**Pull request merged**